examples/proxy/psp22_metadata_upgradeable/lib.rs

```diff
@@ -0,0 +1,46 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+#![feature(min_specialization)]
+
+#[brush::contract]
+pub mod my_psp22 {
+    use brush::contracts::{
+        psp22::extensions::metadata::*,
+        ownable::*,
+    };
+    use ink_prelude::string::String;
+    use ink_storage::traits::SpreadAllocate;
+
+    #[ink(storage)]
+    #[derive(Default, SpreadAllocate, PSP22Storage, PSP22MetadataStorage, OwnableStorage)]
+    pub struct MyPSP22 {
+        #[OwnableStorageField]
+        ownable: OwnableData,
+        #[PSP22StorageField]
+        psp22: PSP22Data,
+        #[PSP22MetadataStorageField]
+        metadata: PSP22MetadataData,
+    }
+
+    impl Ownable for MyPSP22 {}
+    
+    impl PSP22 for MyPSP22 {}
+
+    impl PSP22Metadata for MyPSP22 {}
+
+    impl MyPSP22 {
+        #[ink(constructor)]
+        pub fn new(total_supply: Balance, name: Option<String>, symbol: Option<String>, decimal: u8) -> Self {
+            ink_lang::codegen::initialize_contract(|instance: &mut Self| {
+                instance.initialize(total_supply, name, symbol, decimal)
+            })
+        }
+
+        #[ink(message)]
+        pub fn initialize(&mut self, total_supply: Balance, name: Option<String>, symbol: Option<String>, decimal: u8) {
```

Comment:
`only_owner` modifier should be used=)

Reply:
It also will crash the constructor=) So init owner there to for integration tests